File: backend/app/schemas/image_generation.py
from typing import Any, Literal, Optional

from pydantic import BaseModel, Field, field_validator


class ImageGenerationProvider(BaseModel):
    id: str = "openai_compatible"
    name: str = "OpenAI Compatible"
    type: Literal["openai_compatible"] = "openai_compatible"
    enabled: bool = True
    base_url: str = ""
    api_key: str = ""
    model: str = ""
    size: str = "1024x1024"
    quality: Optional[str] = None
    style: Optional[str] = None
    response_format: Optional[str] = "auto"
    timeout_seconds: int = Field(default=120, ge=5, le=600)


class ImageGenerationDefaults(BaseModel):
    prompt_template: str
    include_recent_context_count: int = Field(default=4, ge=0, le=20)


class ImageGenerationConfigRequest(BaseModel):
    enabled: bool = False
    active_provider_id: str = "openai_compatible"
    providers: list[ImageGenerationProvider] = Field(default_factory=list)
    defaults: Optional[ImageGenerationDefaults] = None


class ImageGenerationTestRequest(BaseModel):
    prompt: str = Field(..., min_length=1, max_length=8000)

    @field_validator("prompt")
    @classmethod
    def validate_prompt(cls, value: str) -> str:
        return value.strip()


class ImageGenerationResult(BaseModel):
    image_url: str
    provider_id: str
    model: Optional[str] = None
    metadata: dict[str, Any] = Field(default_factory=dict)
    prompt: Optional[str] = None
    revised_prompt: Optional[str] = None


class ImageGenerationMessageResponse(BaseModel):
    status: str = "ok"
    image: ImageGenerationResult
    updated_message: dict[str, Any]

File: frontend/src/types/imageGeneration.ts
export interface ImageGenerationProvider {
  id: string;
  name: string;
  type: 'openai_compatible';
  enabled: boolean;
  base_url: string;
  api_key: string;
  model: string;
  size: string;
  quality?: string | null;
  style?: string | null;
  response_format?: 'auto' | 'b64_json' | 'url' | null;
  timeout_seconds: number;
}

export interface ImageGenerationDefaults {
  prompt_template: string;
  include_recent_context_count: number;
}

export interface ImageGenerationConfig {
  enabled: boolean;
  active_provider_id: string;
  providers: ImageGenerationProvider[];
  defaults: ImageGenerationDefaults;
  can_admin?: boolean;
}

export interface ImageGenerationResult {
  image_url: string;
  provider_id: string;
  model?: string | null;
  metadata: Record<string, unknown>;
  prompt?: string | null;
  revised_prompt?: string | null;
}

export interface ImageGenerationTestResponse {
  status: string;
  image: ImageGenerationResult;
}

export interface ImageGenerationMessageResponse {
  status: string;
  image: ImageGenerationResult;
  updated_message: {
    id: string | number;
    role: 'user' | 'assistant' | 'system';
    content: string;
    model?: string | null;
    created_at?: string;
    tokens?: number;
    branch_id?: string | null;
    short_title?: string | null;
  };
}
